Add contextmanager features

JWCook opened the pull request:
This PR adds support for `Limiter.ratelimit()` to be used as a contextmanager (or async contextmanager). To do this I made a new class that implements all the necessary magic methods (`__call__, __enter__, __exit__, __aenter__, __aexit__`). Let me know what you think.

**Questions:**
* Would you prefer if `LimitContextDecorator` stayed in `limiter.py`, or in a separate module?
* Would you be able to change the Code Climate settings for max number of function arguments (currently 4)? It might also be worth increasing the max complexity per function (currently 5).
* It looks like Travis CI stopped triggering builds from PRs: https://travis-ci.com/github/vutran1710/PyrateLimiter/pull_requests Looks like `.travis-ci.yml` hasn't changed, so I'm not sure what the problem is. Unit tests all pass locally.

**Other changes:**
* Reduced complexity per function, so Code Climate should be happy now
    * Edit: Well, complexity is fine, but now it's complaining about having 5 `__init__` arguments 🙄 
* Added some debug logging, which will show up on test failure (or with `pytest -s`)
* Type checkers (mypy and pyright) were complaining about the type hint on `*rates`, so I updated it according to PEP 484: https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values
* Added usage examples to Readme
* Formatted with `black`, bumped line length up to 100 (is that what it previously was with `yapf`?)

vutran1710 commented:
I have refactored the code base https://github.com/vutran1710/PyrateLimiter/pull/26
Can you submit a new PR?
Also, to answer your questions:
- `Would you prefer if LimitContextDecorator stayed in limiter.py, or in a separate module?`. 
I would love to have it in a separate module. Separation saves my eyes from getting tired quickly.

- `Would you be able to change the Code Climate settings for max number of function arguments (currently 4)? It might also be worth increasing the max complexity per function (currently 5)`. 
I have fixed all the complexity issues without changing the setting, so I think thats not an impossible task. Actually, during refactoring the code, I found it quite helpful for pointing those issues out.

- `It looks like Travis CI stopped triggering builds from PRs: https://travis-ci.com/github/vutran1710/PyrateLimiter/pull_requests Looks like .travis-ci.yml hasn't changed, so I'm not sure what the problem is. Unit tests all pass locally.` 
No worry, my bad! It's working now.

JWCook commented:
Sure. Changes have been rebased, I moved my changes to a separate module, and it has 100% coverage now.

Code Climate's pep8 checks were warning about line lengths > 79 characters, so I added a config to `setup.cfg` to increase that to match the flake8 setting. The only remaining warning is for `Limiter.try_aquire()`.
